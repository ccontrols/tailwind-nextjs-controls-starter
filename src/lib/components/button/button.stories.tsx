import React from 'react';
import { Example, Document } from '@component-controls/core';

import { Button, ButtonProps } from './button';

export default {
  title: 'Components/Button',
  component: Button,
  smartControls: {
    include: (control) => control.prop?.parentName === 'ButtonProps',
  },
} as Document;

const Template: Example<ButtonProps> = (args) => <Button {...args}>Button Label</Button>;

export const Primary = Template.bind();
Primary.controls = {
  variant: 'primary',
};

export const Success = Template.bind();
Success.controls = {
  variant: 'success',
};

export const Warn = Template.bind();
Warn.controls = {
  variant: 'warn',
};

export const Danger = Template.bind();
Danger.controls = {
  variant: 'danger',
};

export const FullWidth = Template.bind();
FullWidth.controls = {
  fullWidth: true,
};

export const Medium = Template.bind();
Medium.controls = {
  size: 'medium',
};

export const Small = Template.bind();
Small.controls = {
  size: 'small',
};
