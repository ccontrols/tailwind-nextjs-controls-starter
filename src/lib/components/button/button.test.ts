import * as path from 'path';
import { run, AxeResults } from 'axe-core';
import { reactRunDOM, renderDocument, renderErr } from '@component-controls/test-renderers';
import '@component-controls/jest-axe-matcher';
import { loadConfigurations } from '@component-controls/config';

import { render, act } from '@testing-library/react';

import * as examples from './button.stories';

describe('button', () => {
  const configPath = path.resolve(__dirname, '../../../../.storybook');
  const config = loadConfigurations(configPath);
  let renderedExamples: ReturnType<typeof renderDocument> = [];
  act(() => {
    renderedExamples = renderDocument(examples, config);
  });
  if (!renderedExamples) {
    renderErr();
    return;
  }
  renderedExamples.forEach(({ name, rendered }) => {
    describe(name, () => {
      it('snapshot', () => {
        const { asFragment } = render(rendered);
        expect(asFragment()).toMatchSnapshot();
      });
      it('accessibility', async () => {
        const axeResults = await reactRunDOM<AxeResults>(rendered, run);
        expect(axeResults).toHaveNoAxeViolations();
      });
    });
  });
});
