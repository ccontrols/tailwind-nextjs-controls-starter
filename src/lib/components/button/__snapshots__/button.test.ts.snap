// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button bound Template snapshot 1`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 2`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-green-500 text-white hover:bg-green-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 3`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-yellow-500 text-white hover:bg-yellow-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 4`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-red-500 text-white hover:bg-red-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 5`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8 w-full"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 6`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-4 px-8"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;

exports[`button bound Template snapshot 7`] = `
<DocumentFragment>
  <button
    class="rounded-md focus:outline-none focus:shadow-outline bg-primary-500 text-white hover:bg-primary-600 py-2 px-6"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;
